mio/process/frame_helper.py

```diff
@@ -62,65 +62,149 @@ def split_by_length(self, array: np.ndarray, segment_length: int) -> list[np.nda
 
-        buffer_per_frame = len(serialized_current) // noise_patch_config.buffer_size + 1
+        if config.method == "mean_error":
+            if previous_frame is None:
+                raise ValueError("mean_error requires a previous frame to compare against")
 
-        split_current = self.split_by_length(
-            serialized_current,
-            noise_patch_config.buffer_size // noise_patch_config.buffer_split + 1,
-        )
-        split_previous = self.split_by_length(
-            serialized_previous,
-            noise_patch_config.buffer_size // noise_patch_config.buffer_split + 1,
-        )
+            serialized_current = current_frame.flatten().astype(np.int16)
+            logger.debug(f"Serialized current frame size: {len(serialized_current)}")
 
-        noisy_parts = split_current.copy()
+            serialized_previous = previous_frame.flatten().astype(np.int16)
+            logger.debug(f"Serialized previous frame size: {len(serialized_previous)}")
+
+            split_previous = self.split_by_length(serialized_previous, config.buffer_size)
+            split_current = self.split_by_length(serialized_current, config.buffer_size)
+
+            return self._detect_with_mean_error(split_current, split_previous, config)
+
+        elif config.method == "gradient":
+            return self._detect_with_gradient(current_frame, config)
+        else:
+            logger.error(f"Unsupported noise detection method: {config.method}")
+            raise ValueError(f"Unsupported noise detection method: {config.method}")
+
+    def _detect_with_mean_error(
+        self,
+        split_current: list[np.ndarray],
+        split_previous: list[np.ndarray],
+        config: NoisePatchConfig,
+    ) -> Tuple[bool, np.ndarray]:
+        """
+        Detect noise using mean error between current and previous buffers.
 
+        Returns:
+            Tuple[bool, np.ndarray]: A boolean indicating if the frame is noisy and the noise mask.
+        """
+        noisy_parts = split_current.copy()
         any_buffer_has_noise = False
         current_buffer_has_noise = False
-        for buffer_index in range(buffer_per_frame):
-            for split_index in range(noise_patch_config.buffer_split):
-                i = buffer_index * noise_patch_config.buffer_split + split_index
+
+        logger.debug(f"Config buffer_split: {config.buffer_split}")
+        logger.debug(
+            f"Actual total splits in current: {len(split_current)}, previous: {len(split_previous)}"
+        )
+
+        # buffer_split (splitting each block in smaller segments) cannot be greater than
+        # number of buffers in a frame
+        if config.buffer_split > len(split_current):
+            logger.warning(
+                f"buffer_split ({config.buffer_split}) exceeds total splits "
+                f"({len(split_current)}). Adjusting to {len(split_current)}."
+            )
+            config.buffer_split = len(split_current)
```

Comment:
These parameters shouldn't have this relationship. I took off these and changed `buffer_split` to 10, but it ran finely. Could you double-check if an error really happened? @MarcelMB 
```suggestion
```

Reply:
looks like the test failures here are coming from not detecting that frame 24 which is the one with the very small strip of noise above. 

we might want to do the squared difference to emphasize large differences, again being mindful of dtype :). that might let us get a threshold that separates broken frames from good frames more cleanly

Reply:
Actually, as posted below, this gets detected correctly with the previous threshold and buffer splitting parameters. Seems fairly tricky to decouple code and parameter defaults as it's expected to fail with bad parameters.